src/Maestro/Maestro.Web/Api/v2018_07_16/Controllers/SubscriptionsController.cs

```diff
@@ -109,32 +109,45 @@ public virtual async Task<IActionResult> GetSubscription(Guid id)
         ///   Trigger a <see cref="Subscription"/> manually by id
         /// </summary>
         /// <param name="id">The id of the <see cref="Subscription"/> to trigger.</param>
+        /// <param name="buildId">'bar-build-id' if specified, a specific build is requested</param>
         [HttpPost("{id}/trigger")]
         [SwaggerApiResponse(HttpStatusCode.Accepted, Type = typeof(Subscription), Description = "Subscription update has been triggered")]
         [ValidateModelState]
-        public virtual async Task<IActionResult> TriggerSubscription(Guid id)
+        public virtual async Task<IActionResult> TriggerSubscription(Guid id, [FromQuery(Name = "bar-build-id")] int buildId = 0)
         {
-            Data.Models.Subscription subscription = await TriggerSubscriptionCore(id);
-
-            if (subscription == null)
-                return NotFound();
-
-            return Accepted(new Subscription(subscription));
+            return await TriggerSubscriptionCore(id, buildId);
         }
 
-        protected async Task<Data.Models.Subscription> TriggerSubscriptionCore(Guid id)
+        protected async Task<IActionResult> TriggerSubscriptionCore(Guid id, int buildId = 0)
```

Comment:
Does anyone call this without specifying "buildId"?  Feels like this method doesn't need the default parameter?

Reply:
removed default value